### ISSUE TITLE
Debugger Points to Last Token

### ISSUE THREAD

**Issue by nirvdrum** [oracle/truffleruby#2722]
It looks like our `SourceSection` are wrong or they don't match what the Truffle debugger is expecting. I'm trying to debug a simple Ruby script with VS Code and the helpful arrow to indicate position appears to point at the last token for an expression. Given the following Ruby script:

```ruby
def square(x)
  x * x
end

loop do
  [1, 2, 3, 4].each do |i|
    square(i)
  end
end
```

Set a breakpoint on L6 (the start of the `[1, 2, 3, 4]` array). As you step through the application, the debugger arrow used to indicate where you are in an expression moves around. Stepping over that line will take you into the definition of `loop`, where the "d" in "yield" is highlighted:

![image](https://user-images.githubusercontent.com/12584/188910071-904f27d8-f8d5-471c-a9b9-7a54eb372ddf.png)

Continuing to step through the application, the block argument closing pipe is highlighted rather than the block arguments:

![image](https://user-images.githubusercontent.com/12584/188910246-3f3e7741-b16e-4d15-8d86-a868af0955ae.png)

Stepping into the call to `square`, the "d" in "end" is highlighted:

![image](https://user-images.githubusercontent.com/12584/188910379-65e322c6-a3c7-48b4-840e-8854d97f5f67.png)

**Comment by chrisseaton** [comment]
We simply don't have any column information do we?

**Comment by eregon** [comment]
Indeed, it seems we only give entire lines as SourceSections, because the current parser only gives us that.

Also the RootNode SourceSection for
```ruby
b = proc do |i|
  p square(i)
end
```
seems to only cover the first line, while the body of it covers the first 2 lines (but not the 3rd).

It seems a tooling issue then, if the SourceSection are always whole line(s) then the cursor should be at the very front of the line (or no cursor).
